staging/src/k8s.io/apimachinery/pkg/api/meta/restmapper.go

```diff
@@ -128,6 +128,12 @@ func (m *DefaultRESTMapper) Add(kind schema.GroupVersionKind, scope RESTScope) {
 	m.kindToScope[kind] = scope
 }
 
+// Specifically written for test cases to perform tests on resources.
+func (m *DefaultRESTMapper) AddCustomResource(singular, plural schema.GroupVersionResource) {
```

Comment:
create a discovery.NewRESTMapper and feed in the groups/versions/resources you want to test with as if they had been retrieved via a call to the discovery doc